src/core/dependency-graph.ts

```diff
@@ -12,25 +12,73 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
+import {Deferred} from '../utils';
+
+class DocumentRecord {
+  url: string;
+  dependencies: Set<string>;
+  dependants: Set<string>;
+
+  // Represents the Promise that the dependencies are known, but nothing
+  // about their state (loaded, scanned, etc)
+  _dependenciesDeferred: Deferred<Set<string>>;
```

Comment:
this whole class is private, for the moment. Renamed without `_` to match other fields.